v2/car.go

```diff
@@ -0,0 +1,70 @@
+package car
+
+import (
+	"encoding/binary"
+	"io"
+)
+
+const HeaderBytesLen uint64 = 32
+
+var (
+	// The fixed prefix of a CAR v2, signalling the version number to previois versoions for graceful fail over.
+	PrefixBytes = []byte{
+		0x0a,                                     // unit(10)
+		0xa1,                                     // map(1)
+		0x67,                                     // string(7)
+		0x76, 0x65, 0x72, 0x73, 0x69, 0x6f, 0x6e, // "version"
+		0x02, // uint(2)
+	}
+	// The length of the CAR v2 prefix, i.e. 11 bytes.
+	PrefixBytesLen = uint64(len(PrefixBytes))
+	// Reserved 128 bits space to capture future characteristics of CAR v2 such as order, duplication, etc.
+	EmptyCharacteristics = Characteristics(make([]byte, 16))
+)
+
+type (
+	// Header represents the CAR v2 header/pragma.
+	Header struct {
+		// 128-bit characteristics of this CAR v2 file, such as order, deduplication, etc. Reserved for future use.
+		Characteristics Characteristics
+		// The length of CAR v1 bytes.
+		CarV1Len uint64
```

Comment:
nit: we might want to swap "len" with "size" in all these, because the latter is much more clear in terms of "number of bytes" for most people.

"length" can be a bit more ambiguous: I could interpret it as the string length (number of unicode runes), as the length of the roots list, as the length of the list of blocks...